IPython/html/static/notebook/js/outputarea.js

```diff
@@ -243,11 +243,17 @@ define([
         'text/plain'
     ];
 
-    OutputArea.prototype.validate_mimebundle = function (json) {
-        /**
-         * scrub invalid outputs
-         */
-        var data = json.data;
+    OutputArea.prototype.validate_mimebundle = function (bundle) {
+        /** scrub invalid outputs */
+        if (typeof bundle.data !== 'object') {
+            console.warn("mimebundle missing data", bundle);
+            bundle.data = {};
+        }
+        if (typeof bundle.metadata !== 'object') {
+            console.warn("mimebundle missing metadata", bundle);
+            bundle.metadata = {};
+        }
+        var data = bundle.data;
```

Comment:
doesn't really make a different, but this avoids assigning data twice if it's undefined.